compiler/env/OMRAOTMethodHeader.cpp

```diff
@@ -0,0 +1,183 @@
+   // Now the buffer contains all the data relevant to the
+   // method header.
+   }
+
+OMR::AOTMethodHeader::AOTMethodHeader(uint8_t* serializedMethodData)
+   {
+   // This is an examplar implementation of a creating AOTMethodHeader
+   // class from the raw data received from the storage container
+   // The first sizeof(uintptr_t) bytes
+   // stores the size of the AOTMethodHeader , we will use them
+   // to assert the size computed by the header function matches
+   // the stored size
+
+   size_t storedSize = *((size_t*) serializedMethodData);
+   serializedMethodData += sizeof(size_t);
+
+   // The next sizeof(uint32_t) bytes contain the
+   // size of compiled code
+   self()->setCompiledCodeSize(*((uint32_t*) serializedMethodData));
+   serializedMethodData += sizeof(uint32_t);
+
+   // Right after the compliedCodeSize, the codeSize of bytes
+   // are the code itself, we should record that as a part
+   // of a header
+   self()->setCompiledCodeStart(serializedMethodData);
+
+   // We should skip size of compiled code in bytes
+   // to get to next element of the header
+   serializedMethodData += self()->getCompiledCodeSize();
+
+   // That is relocation size, occupying the next uint32_t bytes
+   self()->setRelocationsSize(*((uint32_t*) serializedMethodData));
+
+   // We should skip the sizeof(uint32_t) in bytes to get
+   // to the relocations information
+   serializedMethodData += sizeof(uint32_t);
+
+   // We take note of the relocationsStart to use the
+   // relocations related information in future
+   self()->setRelocationsStart(serializedMethodData);
+
+   size_t computedSize = self()->sizeOfSerializedVersion();
+
+   TR_ASSERT(computedSize == storedSize, "Stored and Computed MethodHeader sizes mismatch, possible message corruption \n");
+
+   }
+
+size_t
+OMR::AOTMethodHeader::sizeOfSerializedVersion()
+   {
+   // An examplar computation of size  of serialized version
+   // of an AOTMethodHeader, required for memory allocation purposes
+   // The size should generally be composed of
+   // A size_t to hold whole message size,
+   // 2 elements of type uint32_t to store the size of the compiled code and
+   // relocation data
+   // compiledCodeSize of bytes (uint8_t) for compiledcode,
+   // RelocationsSize of bytes (uint8_t) for relocation data.
+   return sizeof(size_t) + 2*sizeof(uint32_t)+self()->getCompiledCodeSize()+self()->getRelocationsSize();
+   }
+
+void
+OMR::AOTMethodHeader::setCompiledCodeStart(uint8_t* codeLocation)
+   {
+   _compiledCodeStart=codeLocation;
+   }
+
+uint8_t*
+OMR::AOTMethodHeader::getCompiledCodeStart()
+   {
+   return _compiledCodeStart;
+   }
+
+void
+OMR::AOTMethodHeader::setCompiledCodeSize(uint32_t codeSize)
+   {
+   _compiledCodeSize = codeSize;
+   }
+
+uint32_t
+OMR::AOTMethodHeader::getCompiledCodeSize()
+   {
+   return _compiledCodeSize;
+   }
+
+void
+OMR::AOTMethodHeader::setRelocationsStart(uint8_t* reloStart)
```

Comment:
same comment about this type...would be better if the parameter has type `TR::RelocationRecordBinaryTemplate` to expose the type expectation to callers .